### ISSUE TITLE
Bug in disambiguation in PGN notation

### ISSUE THREAD

**Issue by lucaferranti** [benckx/elephantchess.io#177]
In the attached screenshot, the third move is

3. Ri2

however, this is ambigous as both chariots could reach the i2 point and it needs to disambiguate, i.e. it should be

3. Rii2

(if I'm not mistaken, PGN should disambiguate first by file and then by rank,hence Rii2 instead of R1i2)

![image](https://github.com/user-attachments/assets/944d5e05-bdc7-45a2-abc1-4466fd177736)

**Comment by lucaferranti** [comment]
for cross-check: here's my [san parser](https://github.com/lucaferranti/elephantops/blob/main/src/san.ts) for xiangqi. Cannot take much credit, I adapted the corresponding lichess library to xiangqi, although bugs are most likely my own ^.^

**Comment by benckx** [comment]
Yes I'm aware of the issue. I made a desambiguation algo to the WXF parser but didn't apply it to other formats. I think I used the same trick as other parsers, i.e. keep all possible interpretations in memory and eliminate the ones that encounter a move that makes no sense or can not be played. With luck at the end there's only 1 valid sequence of moves.

The issue is known but it's possible there was no ticket yet, so I can keep this one.
